api/routes/games.js

```diff
@@ -0,0 +1,245 @@
+const express = require('express');
+const router = express.Router();
+const db = require('../db');
+const dbUtils = require('../utils/db.js');
+const apiUtils = require('../utils/api.js');
+
+const AWS = require('aws-sdk');
+AWS.config.update({
+  region: 'us-east-2',
+  accessKeyId: process.env.AWSAccessKeyId,
+  secretAccessKey: process.env.AWSSecretKey,
+});
+const S3_BUCKET = process.env.Bucket;
+
+/* Get all games */
+router.get('/', (req, res, next) => {
+  db.client.query('SELECT * FROM "Game"', (err, result) => {
+    if (err) {
+      return res.status(400).send(err);
+    }
+    res.status(200).send(result.rows);
+  });
+});
+
+/* Add a game */
+router.post('/', (req, res, next) => {
+  let hostname = req.protocol + '://' + req.headers.host;
+
+  // Map form field names to database field names
+  const fields = {
+    name: 'name',
+    description: 'description',
+    image: '"imgFileName"',
+    publisher: 'publisher',
+    min_players: '"minPlayers"',
+    max_players: '"maxPlayers"',
+    min_playtime: '"minPlaytime"',
+    max_playtime: '"maxPlaytime"',
+    year: 'year',
+    min_age: '"minAge"',
+  };
+
+  // Create array of field names for query
+  let query_fields = Object.keys(fields).map(e => fields[e]);
+
+  // Validate data received
+  if (!("name" in req.body)) {
+    let err = { "Error": "The request object is missing one of the required fields" };
```

Comment:
Maybe we should let them know that `name` is the missing required field?

Reply:
You can either do that as part of the `Error` string or add a `context` key like this

```
let err = {
  'Error': ...,
  'context': {
    field: 'name'
  }
}
```

Totally your call!

Reply:
Updated the error string